erpnext/projects/doctype/timesheet/timesheet.js

```diff
@@ -1,36 +1,9 @@
 // Copyright (c) 2015, Frappe Technologies Pvt. Ltd. and Contributors
 // License: GNU General Public License v3. See license.txt
 
-frappe.ui.form.on("Timesheet", {
-	setup: function(frm) {
-		frm.add_fetch('employee', 'employee_name', 'employee_name');
-		frm.fields_dict.employee.get_query = function() {
-			return {
-				filters:{
-					'status': 'Active'
-				}
-			}
-		}
-
-		frm.fields_dict['time_logs'].grid.get_field('task').get_query = function(frm, cdt, cdn) {
-			var child = locals[cdt][cdn];
-			return{
-				filters: {
-					'project': child.project,
-					'status': ["!=", "Closed"]
-				}
-			}
-		}
-
-		frm.fields_dict['time_logs'].grid.get_field('project').get_query = function() {
-			return{
-				filters: {
-					'company': frm.doc.company
-				}
-			}
-		}
-	},
+{% include 'erpnext/projects/timesheet_common.js' %};
```

Comment:
A couple of things I noticed:
![screenshot-2018-1-10 timesheet creation tool 1](https://user-images.githubusercontent.com/818803/34792617-e92f7c38-f648-11e7-9f98-ed87c55ddb59.png)

After checking 'Bill', The fields with red arrows are not auto filled as usual. Check Timesheet form for the normal behavior.

![screenshot-2018-1-10 timesheet creation tool 2](https://user-images.githubusercontent.com/818803/34792779-74445b2c-f649-11e7-824c-ba1680fd0179.png)
The 'View Timesheets' link is not obvious to the user as a link.

Reply:
Hi @tundebabzy, `billing_hours` get updated only if `Maintain Billing Hours and Working Hours Same on Timesheet` field is checked in HR Settings. Also, the other fields such as `billing_rate` and `costing_rate` get updated if the respective rates are found in Activity Type DocType of that particular Activity.